frontend/javascripts/dashboard/dataset/queries.tsx

```diff
@@ -496,11 +496,14 @@ function diffDatasets(
     .map((id) => newDatasetsDict[id])
     .filter((newDataset) => {
       const oldDataset = oldDatasetsDict[newDataset.id];
-      return !_.isEqualWith(oldDataset, newDataset, (oldValue, newValue, key) => {
-        const didUpgradeToRenamableDS =
-          !("directoryName" in oldValue) && "directoryName" in newValue; // TODO: Can be remove after a few weeks / months.
-        if (key === "lastUsedByUser" || didUpgradeToRenamableDS) {
-          // Ignore the lastUsedByUser timestamp when diffing datasets and migrating datasets to new renamable version.
+      // Ignore updates when migrating to new renamable datasets.
+      const didUpgradeToRenamableDS = ("directoryName" in oldDataset && "directoryName" in newDataset); // TODO: Can be remove after a few weeks / months.
+      if(didUpgradeToRenamableDS) {
+        return false;
+      }
+      return !_.isEqualWith(oldDataset, newDataset, (_oldValue, _newValue, key) => {
+        if (key === "lastUsedByUser") {
+          // Ignore the lastUsedByUser timestamp when diffing datasets.
```

Comment:
@philippotto We have 2 possiblilities of fixing this: Either avoid the invalid access by extracting the check from the inner `isEqualWith` method like done here.